src/digest.rs

```diff
@@ -0,0 +1,23 @@
+use crate::raw;
+use std::os::raw::{c_int};
+
+/// `Digest` is a high-level rust interface to the Valkey module C API 
+/// abstracting away the raw C ffi calls.
+pub struct Digest {
+    pub dig: *mut raw::RedisModuleDigest,
+}
+
+impl Digest {
+    pub const fn new(dig: *mut raw::RedisModuleDigest) -> Self {
+        Self { dig }
+    }
+
+    pub fn key_name(&self) -> *mut raw::RedisModuleString {
+        unsafe { (*self.dig).key }
+    }
+
+    pub fn db_id(&self) -> c_int {
+        unsafe { (*self.dig).dbid }
+    }
```

Comment:
We can implement wrapper functions for the following operations:
```
ValkeyModule_DigestAddStringBuffer
ValkeyModule_DigestAddLongLong
ValkeyModule_DigestEndSequence
ValkeyModule_GetKeyNameFromDigest
ValkeyModule_GetDbIdFromDigest
```

For example, we can add a  function named `add_long_long`.
This function will look something like:
```
pub fn add_long_long(&mut self, ll: i64) {
    unsafe {
        // Make a call to raw::RedisModule_DigestAddLongLong API and mutatively pass "self.dig" and "ll" c_longlong type
    }
}
```

Reply:
Thanks for the feedback Karthik - [updated here](https://github.com/valkey-io/valkeymodule-rs/pull/102/commits/ff96f1b2583c3ec2896e405b01bc8854aa241316)